pkg/kwokctl/runtime/util.go

```diff
@@ -43,3 +49,47 @@ func ExpandVolumesHostPaths(volumes []internalversion.Volume) ([]internalversion
 	}
 	return result, nil
 }
+
+// GetLogVolumes returns volumes for Logs and ClusterLogs resource.
+func GetLogVolumes(ctx context.Context) ([]internalversion.Volume, error) {
+	logs := config.FilterWithTypeFromContext[*internalversion.Logs](ctx)
+	clusterLogs := config.FilterWithTypeFromContext[*internalversion.ClusterLogs](ctx)
+
+	// Mount log dirs
+	mountDirs := make(map[string]struct{})
+	for _, log := range logs {
+		for _, l := range log.Spec.Logs {
+			abs, err := path.Expand(l.LogsFile)
+			if err != nil {
+				return nil, err
+			}
+			mountDirs[filepath.Dir(abs)] = struct{}{}
+		}
+	}
+
+	for _, cl := range clusterLogs {
+		for _, l := range cl.Spec.Logs {
+			abs, err := path.Expand(l.LogsFile)
+			if err != nil {
+				return nil, err
+			}
+			mountDirs[filepath.Dir(abs)] = struct{}{}
+		}
+	}
+
+	volumes := make([]internalversion.Volume, 0, len(mountDirs))
+	i := 0
+	for dir := range mountDirs {
+		dirPath := strings.TrimPrefix(dir, "/var/components/controller")
```

Comment:
This is the part I am unsure of. We can probably add strip prefix only in case when runtime is kind because we know that the prefix is going to be added, and it is going to be used in template.

The issue is that to use extra volume mounts, we add `/var/components/controller` to all paths.
Since we need correct configuration for logs, we expand the path. I agree that the server should not inspect the runtime from configuration, so that is why this change was necessary. Ideally it should not interfere with the binary runtime unless the actual path there is this constant. Because of that, It might be wise to use check for the runtime but I wanted to check first, since this approach might be flawed in a way I don't know :smile: